### PR TITLE
feat: watermark

### PR DESCRIPTION
**Pull request by ybzky**
close #xxx

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/watermark/src/commands/operations/OpenWatermarkPanelOperation.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ICommand } from '@univerjs/core';
+import { CommandType, LocaleService } from '@univerjs/core';
+import { ISidebarService } from '@univerjs/ui';
+import { WATERMARK_PANEL, WATERMARK_PANEL_FOOTER } from '../../common/const';
+
+export const OpenWatermarkPanelOperation: ICommand = {
```
Comment (wzhudev) [inline packages/watermark/src/commands/operations/OpenWatermarkPanelOperation.ts:22]: Name of this file does not follow our naming convension.

File: packages/watermark/src/services/watermarkService.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * limitations under the License.
+ */
+
+import type { Nullable } from '@univerjs/core';
+import type { IWatermarkConfigWithType } from '../common/type';
+import { Disposable, ILocalStorageService, Inject } from '@univerjs/core';
+import { Subject } from 'rxjs';
+import { UNIVER_WATERMARK_STORAGE_KEY } from '../common/const';
+
+export class UniverWatermarkService extends Disposable {
```
Comment (wzhudev) [inline packages/watermark/src/services/watermarkService.ts:23]: Naming convension.

**Review by wzhudev — CHANGES_REQUESTED** [review]
README of this plugin is empty.

File: packages/sheets-ui/src/services/sheets-render.service.ts [hunk 3]
```diff
@@ -87,6 +87,7 @@ export class SheetsRenderService extends RxDisposable {
         this._renderManagerService.created$.subscribe((renderer) => {
             if (renderer.unitId === unitId) {
                 renderer.engine.getCanvas().setId(SHEET_MAIN_CANVAS_ID);
+                renderer.engine.getCanvas().getContext().setId(SHEET_MAIN_CANVAS_ID);
```
[the remark this answers is not in the document]
Comment (ybzky) [reply packages/sheets-ui/src/services/sheets-render.service.ts:90]: here only set id to the main canvas

File: packages/watermark/src/services/watermark.service.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+import type { Nullable } from '@univerjs/core';
+import type { IWatermarkConfigWithType } from '../common/type';
+import { Disposable, ILocalStorageService, Inject } from '@univerjs/core';
+import { Subject } from 'rxjs';
+import { UNIVER_WATERMARK_STORAGE_KEY } from '../common/const';
+
+export class WatermarkService extends Disposable {
+    private _updateConfig$ = new Subject<Nullable<IWatermarkConfigWithType>>();
```
Comment (wzhudev) [inline packages/watermark/src/services/watermark.service.ts:24]:
```suggestion
    private readonly _updateConfig$ = new Subject<Nullable<IWatermarkConfigWithType>>();
```

Comment (wzhudev) [reply packages/watermark/src/services/watermark.service.ts:24]: And these observables should `complete` on disposing.

**Review by wzhudev — APPROVED** [review]
LGTM. An e2e visual comparison test would be better.

**Pull request merged**